### ISSUE TITLE
Extensions don't load when VSCode is launched via app/bin/code on command line

### ISSUE THREAD

**Issue by z-aki** [microsoft/vscode#128723]
Issue Type: <b>Bug</b>

VS Code version: Code 1.58.1 (2aeda6b18e13c4f4f9edf6667158a6b8d408874b, 2021-07-13T06:22:33.013Z)
OS version: Darwin x64 18.7.0
Restricted Mode: No

Extensions don't load when VSCode is launched via app/bin/code on command line
Issue happens on fresh installation (i.e. deleting ~/Application Support/Code & saved state etc).

Steps to Reproduce:

0. Move VScode app to a folder other than `/Applications`.  the bug doesn't happen if vscode is in /Applications.
1. Run `"/Path_to/Visual Studio Code.app/Contents/Resources/app/bin/code" path_to_any_directory_doesnt_matter`
2. Search for any extension in the marketplace.
3. See that the blue progress bar goes on forever. Internet is working fine as we'll see in step 6.
4. Quit it and run Visual studio code by double clicking the app icon. 
5. Repeat step 2.
6. See that extension results show up.

**Comment by sandy081** [comment]
Can you please check following?

- Run Command - `Developer: Toggle Shared Process` and see if it can be opened?

**Comment by z-aki** [comment]
@sandy081  yes the developer window opens if I run that command
Also, I just added another step 0, it doesn't happen if vscode is in /Applications.

**Comment by z-aki** [comment]
Or maybe it's because:
- have a clean slate
- put vscode in /Applications, launch it in any way and edit a few settings.
- *Move* vscode to another folder outside /Applications
- launch it from command line etc as given in the top box.

**Comment by sandy081** [comment]
@joaomoreno / @deepak1556 any insights ?

**Comment by deepak1556** [comment]
I am unable to repro with the above steps, @z-aki please run `Developer: Set Log Level` and set it to trace. After you encounter the issue run `Developer: Show Logs Folder` and attach the logs from it.

**Comment by z-aki** [comment]
@deepak1556 https://github.com/z-aki/VSClogs
Is there a way to have this setting persist between sessions ? so that trace can take effect from launch, instead of when I set it to trace.

**Comment by deepak1556** [comment]
Thanks for the logs, everything looks good in there. Can you provide the output of launching with `--verbose` for the custom folder setup with performing the extension search.

> Is there a way to have this setting persist between sessions ? so that trace can take effect from launch, instead of when I set it to trace.

I am not aware of it, @sandy081 do you know how persist the log level ?

**Comment by z-aki** [comment]
It seems verbose implies trace. So here's everything https://github.com/z-aki/VSClogs/tree/verbose

**Comment by sandy081** [comment]
> do you know how persist the log level ?

It can be done in the `args.json` file. You can open it using the command `Developer: Configure Runtime Arguments` and set
`log-level` property.

**Comment by z-aki** [comment]
Thanks!
Added verbose + trace. https://github.com/z-aki/VSClogs/tree/verbose_trace

**Comment by deepak1556** [comment]
Runtime log looks fine, can you follow the steps below to get a network log.

* Open code with this additional flag `--log-net-log=<some-absolute-path-to>/netlog.json`
* Ensure that the extension search fails
* Quit the application
* Attach netlog.json file generated at the path

**Comment by z-aki** [comment]
https://github.com/z-aki/VSClogs/tree/netlog_verbose_trace

**Comment by deepak1556** [comment]
Thanks for the log

@sandy081 I only see a single request for the update check in the net log, so looks like the extension search didn't even hit the network layer. Can we get additional logs in the extension search code path to see where it failed in the runtime ?

**Comment by z-aki** [comment]
Okay the issue resolved. I am using a firewall. Whenever a process tries to connect to the internet, the fw pops up and waits for my confirmation whether to allow or block the request.
When code is launched via CLI, code helper's request never showed up.
So I launched it from app icon, auto update request request showed up, I approved it. Also, an extension install related request (judging from the server url) showed up, I approved it.
Then launched code via cli, and now search works.
